src/main/java/uk/gov/hmcts/reform/et/syaapi/service/SendNotificationService.java

```diff
@@ -51,4 +68,73 @@ public CaseDetails updateSendNotificationState(String authorization, SendNotific
         );
     }
 
+    /**
+     * Adds a pseResponse to a notification.
+     *
+     * @param authorization - authorization
+     * @param request - request containing the response, and the notification details
+     * @return the associated {@link CaseDetails}
+     */
+    public CaseDetails addResponseSendNotification(String authorization, SendNotificationAddResponseRequest request) {
+        StartEventResponse startEventResponse = caseService.startUpdate(
+            authorization,
+            request.getCaseId(),
+            request.getCaseTypeId(),
+            CaseEvent.UPDATE_NOTIFICATION_RESPONSE
+        );
+
+        CaseData caseData = EmployeeObjectMapper
+            .mapRequestCaseDataToCaseData(startEventResponse.getCaseDetails().getData());
+        var sendNotificationTypeItem =
+            caseData.getSendNotificationCollection()
+                .stream()
+                .filter(notification -> notification.getId().equals(request.getSendNotificationId()))
+                .findFirst();
+        if (sendNotificationTypeItem.isEmpty()) {
+            throw new IllegalArgumentException("SendNotification Id is incorrect");
+        }
+
+        SendNotificationType sendNotificationType = sendNotificationTypeItem.get().getValue();
+
+        var pseRespondCollection = sendNotificationType.getRespondCollection();
+        if (CollectionUtils.isEmpty(pseRespondCollection)) {
+            sendNotificationTypeItem.get().getValue().setRespondCollection(new ArrayList<>());
+        }
+
+        PseResponseType pseResponseType = request.getPseResponseType();
+        pseResponseType.setDate(TseApplicationHelper.formatCurrentDate(LocalDate.now()));
+        pseResponseType.setFrom(CLAIMANT);
+
+        if (request.getSupportingMaterialFile() != null) {
```

Comment:
small comment: maybe swap if and else conditions?

Reply:
Maybe even just take out the pseResponseType.setHasSupporting into a ternary and then just one if(!null) {}